src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,288 @@
+    verifyFirstRowModelEqualToExample(exampleNotificationOne);
+  });
+
+  it('should display error message when failed to create notification', () => {
+    component.notificationEditFormModel = exampleNotificationEditModel;
+    jest.spyOn(notificationService, 'createNotification').mockReturnValue(throwError({
+      error: {
+        message: 'This is the error message.',
+      },
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showErrorToast')
+      .mockImplementation((args: string) => {
+        expect(args).toEqual('This is the error message.');
+      });
+
+    component.addNewNotificationHandler();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(component.notificationsTableRowModels.length).toEqual(0);
+  });
+
+  it('should display error message when notification failed to load', () => {
+    component.hasNotificationLoadingFailed = false;
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(throwError({
+      error: {
+        message: 'This is the error message.',
+      },
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showErrorToast')
+      .mockImplementation((args: string) => {
+        expect(args).toEqual('This is the error message.');
+      });
+
+    component.loadNotifications();
+
+    expect(spy).toBeCalled();
+    expect(component.hasNotificationLoadingFailed).toBeTruthy();
+  });
+
+  it('should display warning when attempts to edit another notification when form is open', async () => {
+    const promise: Promise<void> = Promise.resolve();
+    const modalSpy: SpyInstance = jest.spyOn(simpleModalService, 'openConfirmationModal')
+        .mockReturnValue(createMockNgbModalRef({}, promise));
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(of({
+      notifications: [exampleNotificationOne, exampleNotificationTwo],
+    }));
+    component.initNotificationEditFormModel();
+    component.loadNotifications();
+    component.loadNotificationEditForm(component.notificationsTableRowModels[0].notification);
+    fixture.detectChanges();
+    expect(component.isNotificationEditFormExpanded).toBeTruthy();
+
+    const expectedMessage: string = component.notificationsTableRowModels[0].notification.message;
+
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[0].notification);
+    component.notificationEditFormModel.message = 'new message';
+    expect(component.notificationEditFormModel.message).toEqual('new message');
+
+    // attempts to load edit form for another notification
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[1].notification);
+    await promise;
+    expect(component.notificationsTableRowModels[0].notification.message).toEqual(expectedMessage);
+    expect(component.notificationEditFormModel.message)
+        .toEqual(component.notificationsTableRowModels[1].notification.message);
```

Comment:
Not very clear what this is doing, given test description `'should display warning when attempts to edit another notification when form is open'`, perhaps can add a comment for clarity.

Reply:
Sure, line 251 tests if the changes to previous notification is discarded, while line 252-253 tests the edit form model loads information of the new notification

Reply:
Comments have been added in the new commit. Thank you!